lib/Parser/Scan.cpp

```diff
@@ -1934,16 +1936,19 @@ tokens Scanner<EncodingPolicy>::ScanCore(bool identifyKwds)
             }
         case '(': Assert(chType == _C_LPR); token = tkLParen; break;
         case ')': Assert(chType == _C_RPR); token = tkRParen; break;
-        case ',': Assert(chType == _C_CMA); token = tkComma; break;
+        case ',': Assert(chType == _C_CMA); token = tkComma;  break;
         case ';': Assert(chType == _C_SMC); token = tkSColon; break;
         case '[': Assert(chType == _C_LBR); token = tkLBrack; break;
         case ']': Assert(chType == _C_RBR); token = tkRBrack; break;
-        case '~': Assert(chType == _C_TIL); token = tkTilde; break;
-        case '?': Assert(chType == _C_QUE); token = tkQMark; break;
-        case '{': Assert(chType == _C_LC); token = tkLCurly; break;
+        case '~': Assert(chType == _C_TIL); token = tkTilde;  break;
+        case '?': Assert(chType == _C_QUE); token = tkQMark;  break;
+        case '{': Assert(chType == _C_LC);  token = tkLCurly; break;
 
+        // ES 2015 11.3 Line Terminators
         case '\r':
         case '\n':
+        case kchLS:
+        case kchPS:
```

Comment:
It would appear that this was already handled by the default case at the top of the switch with a goto LNewLine.

However, if the scanner is in utf8 mode these characters will not be handled because this switch is operating only on the first part of the character via ReadFirst().  Is this broken in utf8?